inspirehep/modules/editor/views.py

```diff
@@ -225,3 +228,22 @@ def _simplify_ticket_response(ticket):
         owner=ticket['Owner'],
         date=ticket['Created'],
         link=ticket['Link'])
+
+
+@blueprint.route('/upload', methods=['POST'])
+@editor_use_api_permission.require(http_exception=403)
+def upload_files():
+    # check if the post request has the file part
+    if 'file' not in request.files:
+        return jsonify(success=False, message='No file part')
+    attachment = request.files['file']
+    # if user does not select file, browser also
+    # submit a empty part without filename
+    if attachment:
+        filedir = current_app.config['RECORD_EDITOR_FILE_UPLOAD_FOLDER']
+        if not os.path.isdir(filedir):
+            os.makedirs(filedir)
+        filename = secure_filename(attachment.filename)
```

Comment:
seems like if you upload a file with same name, it'll override. I think file name should be generated or sth.

Reply:
@harunurhan check it again. I updated with check for file existence.